spec/unit/fibered_mysql2_adapter_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require_relative '../../lib/active_record/connection_adapters/fibered_mysql2_adapter'
+
+RSpec.describe FiberedMysql2::FiberedMysql2Adapter do
+  let(:client) { double(Mysql2::EM::Client) }
+  let(:logger) { Logger.new(STDOUT) }
+  let(:options) { [] }
+  let(:config) { {} }
+
+  subject { FiberedMysql2::FiberedMysql2Adapter.new(client, logger, options, config) }
+
+  before do
+    allow(client).to receive(:query_options) { {} }
+    allow(client).to receive(:escape) { |query| query }
+    query_args = []
+    stub_mysql_client_result = Struct.new(:fields, :to_a).new([], [])
+    expect(client).to receive(:query) do |*args|
+      query_args << args
+      stub_mysql_client_result
+    end.at_least(1).times
+    allow(client).to receive(:server_info).and_return({ version: "5.7.27" })
+  end
+
+  it 'returns a FiberedMysql2Adapter' do
+    expect(subject).to be_instance_of(FiberedMysql2::FiberedMysql2Adapter)
+  end
```

Comment:
Tests like this can be simplified in Rspec to 
```suggestion
  it { should be_instance_of FiberedMysql2::FiberedMysql2Adapter }
```